### ISSUE TITLE
[DACP] Remediate MSP Accessibility Issues

### ISSUE THREAD

**Issue by KaroEngstrom** [cityofaustin/atd-data-tech#27688]
# Wed 4/1/26

* [AccTrack](https://atd.knack.com/dts#accessibility-items/) (Accessibility Tracker) in DTS Portal 
* Filter to the app in question 
* On the Item Details page, update Issues by inline-editing the row for the issue (Status, Comment, and Attachment are inline-editable) 
    * Update Status to Resolved when remediated 
    * Add Comment if you consider it helpful 
    * Add Attachment if you consider it helpful 
* Note that some issues (page zoom, notably) are systemic to the platform and we are waiting to get help from vendors to solve those

**Comment by susannegov** [comment]
# 04/01
- Added OA account in dummy

Going through the SiteImprove
- Update dropdown menu code for accessibility. Will submit a PR to address
- ~~There is a container list in empty being flagged and I strongly believe it is a bug. A persistent empty notification that if you press the F8 keyboard it highlights the empty notification. This is not a custom code related as I was seeing this in my sandbox as flagged by siteimprove~~
   - ~~Sandbox public version (with no custom code): https://atd.knack.com/sandbox-sg#search-license-plate/~~
      - ~~You can test this by pressing F8 and seeing the border highlight~~
   - ~~This could be flagged as vendor. Not all pages have it(?)~~
   - EDIT: This turned out to be from a browser extension I installed from Scribe, not Knack. After disabling extension it passes the check.

<img width="1027" height="298" alt="Image" src="https://github.com/user-attachments/assets/64d6f051-70c9-4266-b278-cbe823f5ac06" />

**Comment by susannegov** [comment]
# 04/02
- Remediate list role by removing `<i>` styling in Chauffeur https://github.com/cityofaustin/atd-data-tech/issues/22783
- If using Date/Time or Multichoice as a drop down menu the label field will not work
- If using Title/Copy in the Details label it will display an empty `<h3>` tag. Workaround in CSS to detect **empty** h3 tags and hide them from screen readers for print pages

```
#kn-scene_83 h3:empty,
#kn-scene_65 h3:empty {
  display: none !important;
}
```

- Resolved container list in widget was from my browser extension
- Update dropdown menu code to Rich Text instead of menu views to fix empty containers
- Moved headings from Copy to Title to fix empty headings for text that is a heading style in Chauffeur

## Accessibility Notes
- The mobile navigation icon is flagged for no label "Button missing text alternative"

<img width="475" alt="Image" src="https://github.com/user-attachments/assets/6e65f2de-deee-4524-9144-d9f3661c04ef" />

- Fields I noticed on a systematic level "Form field missing a label" in SiteImprove
   - Date/Time
      - In next Gen this isn't flagged as they use `aria-label="Date Time Input"` 

<img width="735"  alt="Image" src="https://github.com/user-attachments/assets/5ddb6afe-8919-4f12-9508-4f68fa7315f1" />

   - Multichoice with specific layout:
      - Dropdown - One Selection
      - Dropdown - Multiple Selections
         - If there is an "x" to remove  a selection SiteImprove would show "Link missing a text alternative"
         - Connection field types might be affected by this

<img width="757" alt="Image" src="https://github.com/user-attachments/assets/95ac3a24-f70d-41e6-910e-dfeeafa8c5f0" />

- Might require `arialabeledby` since these are tied to inputs
   - [Documentation and example](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)

## Option 1 `aria-labelledby`
You have to reference the id that specifically that already has a label. If a a label already exists it will reference it.
```
<div class="kn-input kn-input-date_time control" id="kn-input-field_34" data-input-id="field_34">
<label for="field_34" class="label kn-label"><span>Date of Birth</span> <span class="kn-required">*</span></label>
<input id="view_38-field_34" name="date" type="text" aria-labelledby="kn-input-field_34">
```

## Option 2 `aria-label`
Uses descriptions
```
<label for="field_34" class="label kn-label"><span>Date of Birth</span> <span class="kn-required">*</span></label>
<input id="view_38-field_34" name="date" type="text" aria-label="Input Date of Birth">
```

## To do
- Submit PR of needed CSS changes
- Writeup of suspected vendor-related accessibility issues and workarounds

**Comment by susannegov** [comment]
# 04/03
## Print Chauffeur Long/Short
- Added Rich Text and HTML to labels
- Removed CSS for print chauffeur pages
- Changed font color from `#FF0000` to `#EE0000` for red text for color contrast in print page

**Comment by susannegov** [comment]
# 04/08
- Created PR https://github.com/cityofaustin/atd-knack/pull/255

## To do
Once PR is approved update
- [x] In-form dropdown menu https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-LzNSSXajDFpzOv2IhIv/knack-code/looks/in-form-dropdown-menu-buttons
- [x] Disable Trigger Buttons https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-LzNSSXajDFpzOv2IhIv/knack-code/functionality/disable-trigger-buttons

**Comment by susannegov** [comment]
# 04/10
- Updated Gitbook pages above
- Update AccTracker to "Vendor Pending" on "form fields missing label"

# To do
All builder related updates for accessibility are done. The only ones not completed is vendor pending ones such as:
- Page Zoom
- "Form field missing label"

**Comment by susannegov** [comment]
# 05/06
- Checked pages
- Closing Issue for nonvendor related issues
